client/state/data-layer/wpcom/sites/rewind/restores/index.js

```diff
@@ -0,0 +1,76 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { errorNotice } from 'state/notices/actions';
+import { registerHandlers } from 'state/data-layer/handler-registry';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { REWIND_RESTORE_DISMISS_PROGRESS } from 'state/action-types';
+
+/**
+ * Mark a specific restore record as dismissed.
+ * This has the effect that subsequent calls to /sites/%site_id%/rewind/restores won't return the restore.
+ *
+ * @param   {object}   action   Changeset to update state.
+ * @returns {object}          The dispatched action.
+ */
+export const dismissRestore = action =>
+	http(
+		{
+			method: 'POST',
+			apiNamespace: 'wpcom/v2',
+			path: `/sites/${ action.siteId }/rewind/restores/${ action.restoreId }`,
+			body: {
+				dismissed: true,
+			},
+		},
+		action
+	);
+
+/**
+ * On successful dismiss, the card will be removed and we don't need to do anything further.
+ * If request succeeded but restore couldn't be dismissed, a notice will be shown.
+ *
+ * @param {object}   action   Changeset to update state.
+ * @param {object}     data     Description of request result.
+ */
+export const restoreSilentlyDismissed = ( action, data ) =>
+	! data.dismissed
+		? errorNotice( translate( 'Dismissing restore failed. Please reload and try again.' ) )
+		: null;
+
+/**
+ * If a dismiss request fails, an error notice will be shown.
+ *
+ * @returns {function} The dispatched action.
+ */
+export const restoreDismissFailed = () =>
+	errorNotice( translate( 'Dismissing restore failed. Please reload and try again.' ) );
+
+/**
+ * Parse and merge response data for restore dismiss result with defaults.
+ *
+ * @param   {object} data   The data received from API response.
+ * @returns {object} Parsed response data.
+ */
+const fromRestoreDismiss = data => ( {
+	restoreId: +data.restore_id,
```

Comment:
`parseInt` is probably better here.

Reply:
Agreed, but unary is used throughout calypso so I stayed with convention. I am happy that you said this, I'll change it :-)